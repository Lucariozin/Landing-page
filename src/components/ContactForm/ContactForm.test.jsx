import { ContactForm } from '.';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

describe('<ContactForm />', () => {
  it('should render ContactForm correctly', () => {
    const { container } = render(<ContactForm />);

    const button = container.querySelector('button');
    const input = container.querySelector('input[placeholder]');

    expect(screen.getAllByPlaceholderText(/Your/i)).toHaveLength(4);
    expect(button).toHaveTextContent(/Send message/i);
    expect(screen.getByAltText(/Imagem representando formas de contato/i));

    userEvent.type(input, 'test');
    expect(input).toHaveValue('test');
  });

  it('should to match snapshot', () => {
    const { container } = render(<ContactForm />);
    expect(container).toMatchSnapshot();
  });
});
