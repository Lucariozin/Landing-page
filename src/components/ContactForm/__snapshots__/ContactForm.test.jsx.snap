// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContactForm /> should to match snapshot 1`] = `
.c0 {
  width: 100%;
  margin: 8rem 0;
  display: grid;
  grid-template-columns: 1.5fr 2fr;
  gap: 2rem;
}

.c2 {
  width: 100%;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c8 {
  font-size: 1.6rem;
  padding: 0.5rem 1rem;
  border-radius: 0.5rem;
  min-width: 30rem;
  width: 100%;
}

.c4 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 2rem;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c6 {
  font-size: 1.6rem;
  padding: 0.5rem 1rem;
  border-radius: 0.5rem;
  max-width: 15rem;
  width: 100%;
}

.c9 {
  margin: 2rem 0;
  padding: 1rem;
  font-size: 1.6rem;
  border-radius: 0.5rem;
  resize: none;
  max-width: 64rem;
  width: 100%;
  height: 15rem;
}

.c10 {
  margin-top: 2rem;
  max-width: 15rem;
  font-size: 1.6rem;
  font-weight: 600;
  border-radius: 0.5rem;
  border: 0.2rem solid var(--white);
  padding: 1rem;
  -webkit-transition: 300ms all ease-in-out;
  transition: 300ms all ease-in-out;
}

.c10:hover {
  -webkit-filter: brightness(0.8);
  filter: brightness(0.8);
  color: var(--white);
  background: transparent;
  border: 0.2rem solid var(--white);
}

@media (max-width:950px) {
  .c0 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
  }
}

@media (max-width:950px) {
  .c1 {
    display: none;
  }
}

@media (max-width:550px) {
  .c4 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-align-items: flex-start;
    -webkit-box-align: flex-start;
    -ms-flex-align: flex-start;
    align-items: flex-start;
  }

  .c4 .c7 {
    min-width: 10rem;
  }
}

<div>
  <div
    class="c0"
    id=""
  >
    <div
      class="c1"
    >
      <img
        alt="Imagem representando formas de contato"
        class="c2"
        src="assets/img/contact.svg"
      />
    </div>
    <form
      class="c3"
    >
      <div
        class="c4"
      >
        <label
          class="c5"
        >
          First name
          <input
            class="c6"
            placeholder="Your first name"
            type="text"
            value=""
          />
        </label>
        <label
          class="c5"
        >
          Last name
          <input
            class="c6"
            placeholder="Your last name"
            type="text"
            value=""
          />
        </label>
        <label
          class="c5"
        >
          E-mail
          <input
            class="c7 c8"
            placeholder="Your e-mail"
            type="email"
            value=""
          />
        </label>
      </div>
      <textarea
        class="c9"
        placeholder="Your message"
      />
      <button
        class="c10"
        type="submit"
      >
        Send message
      </button>
    </form>
  </div>
</div>
`;
